### PR TITLE
Improve reliability of tests

### PR DESCRIPTION
**Pull request by owais**
# Description

Update some tests to use more deterministic methods of testing in memory
spans. This helps the core repo pass tests after adding Windows to CI
matrix.

## Type of change

Please delete options that are not relevant.

- [x] Dev/Tooling enchancement

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

This PR has been split out of #531 in order to not block the core Windows PR. We can merge core Windows PR along with this one and continue to work on contrib Windows.

- [x] Updated Tests

# Does This PR Require a Core Repo Change?

- [x] Yes. - Link to PR: https://github.com/open-telemetry/opentelemetry-python/pull/1889
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py [hunk 1]
```diff
@@ -270,10 +263,11 @@ def test_dsl_create(self, request_mock):
         client = Elasticsearch()
         Article.init(using=client)
 
-        spans = self.get_ordered_finished_spans()
+        spans = self.get_finished_spans()
         self.assertEqual(2, len(spans))
-        span1, span2 = spans
-        self.assertEqual(span1.name, "Elasticsearch/test-index")
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py:276]: Why delete this check?

File: tox.ini [hunk 2]
```diff
@@ -122,7 +122,8 @@ envlist =
     py3{6,7,8,9}-test-instrumentation-grpc
 
     ; opentelemetry-instrumentation-sqlalchemy
-    py3{6,7,8,9}-test-instrumentation-sqlalchemy{11,14}
+    py3{6,7}-test-instrumentation-sqlalchemy{11}
```
Comment (aabmass) [inline tox.ini:125]: Can you add a comment explaining why we do this?

**Review by aabmass — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (owais) [reply tox.ini:125]: https://github.com/open-telemetry/opentelemetry-python-contrib/pull/531/files#r696214503

Will add a comment to the file.

**Review by ocelotl — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py (hunk 1 above)
Comment (owais) [reply instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py:276]: LMK if that makes sense. I can follow up with another PR.

**Pull request merged**